operator/controllers/helmrelease_controller.go

```diff
@@ -36,24 +37,30 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 )
 
-// The HelmReleaseFinalizer allows the controller to clean up the associated
-// release before the HelmRelease resource is deleted.
+// HelmReleaseFinalizer allows the controller to clean up the associated release
+// before the HelmRelease resource is deleted.
 const HelmReleaseFinalizer = "HelmReleaseFinalizer"
 
-var errNotImplemented = errors.New("not implemented")
-
 type ChartDownloader interface {
 	Download(ctx context.Context, chart string, version string) (*chart.Chart, error)
 }
 
 type HelmClient interface {
+	ReleaseCRUD
+	ReleaseStatuser
+}
+
+type ReleaseCRUD interface {
```

Comment:
I liked separating the `HelmClient` methods used by the reconciler vs the release manager, though there's probably a better name for this interface.